src/EditorFeatures/Core/Classification/Semantic/AbstractSemanticOrEmbeddedClassificationViewTaggerProvider.cs

```diff
@@ -67,36 +97,193 @@ protected sealed override ITaggerEventSource CreateEventSource(ITextView textVie
             TaggerEventSources.OnGlobalOptionChanged(_globalOptions, ClassificationOptionsStorage.ClassifyObsoleteSymbols));
     }
 
-    protected sealed override Task ProduceTagsAsync(
+    protected sealed override async Task ProduceTagsAsync(
         TaggerContext<IClassificationTag> context, DocumentSnapshotSpan spanToTag, CancellationToken cancellationToken)
     {
         var document = spanToTag.Document;
         if (document == null)
-            return Task.CompletedTask;
+            return;
 
         // Attempt to get a classification service which will actually produce the results.
         // If we can't (because we have no Document, or because the language doesn't support
         // this service), then bail out immediately.
         var classificationService = document.GetLanguageService<IClassificationService>();
         if (classificationService == null)
-            return Task.CompletedTask;
+            return;
 
         // The LSP client will handle producing tags when running under the LSP editor.
         // Our tagger implementation should return nothing to prevent conflicts.
         var workspaceContextService = document.Project.Solution.Services.GetRequiredService<IWorkspaceContextService>();
         if (workspaceContextService?.IsInLspEditorContext() == true)
-            return Task.CompletedTask;
+            return;
 
         // If the LSP semantic tokens feature flag is enabled, return nothing to prevent conflicts.
         var isLspSemanticTokensEnabled = _globalOptions.GetOption(LspOptionsStorage.LspSemanticTokensFeatureFlag);
         if (isLspSemanticTokensEnabled)
-            return Task.CompletedTask;
+            return;
+
+        var classificationOptions = _globalOptions.GetClassificationOptions(document.Project.Language) with
+        {
+            FrozenPartialSemantics = context.FrozenPartialSemantics,
+        };
 
-        var classificationOptions = _globalOptions.GetClassificationOptions(document.Project.Language);
-        return ClassificationUtilities.ProduceTagsAsync(
-            context, spanToTag, classificationService, _typeMap, classificationOptions, _type, cancellationToken);
+        await ProduceTagsAsync(
+            context, spanToTag, classificationService, classificationOptions, cancellationToken).ConfigureAwait(false);
     }
 
-    protected override bool TagEquals(IClassificationTag tag1, IClassificationTag tag2)
-        => tag1.ClassificationType.Classification == tag2.ClassificationType.Classification;
+    public async Task ProduceTagsAsync(
```

Comment:
these are just method moves from a utilities class far awqay. they were only called here (and referred to tagger types).  so i just moved them all back over.  they are otherwise unchanged.